<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SongEdit</class>
 <widget class="QDialog" name="SongEdit">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>650</width>
    <height>750</height>
   </rect>
  </property>
  <property name="acceptDrops">
   <bool>true</bool>
  </property>
  <property name="windowTitle">
   <string>Untitled - Song Editor</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color:white</string>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QGridLayout" name="mainLayout">
   <item row="1" column="0" colspan="2">
    <widget class="QGroupBox" name="horizontalGroupBox">
     <property name="font">
      <font>
       <bold>true</bold>
      </font>
     </property>
     <property name="title">
      <string>Location</string>
     </property>
     <layout class="QHBoxLayout" name="fileNameLayout">
      <property name="topMargin">
       <number>1</number>
      </property>
      <item>
       <widget class="QLabel" name="fileNameLabel">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>File Name:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
        </property>
        <property name="buddy">
         <cstring>fileNameEdit</cstring>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="fileNameEdit">
        <property name="readOnly">
         <bool>true</bool>
        </property>
        <property name="placeholderText">
         <string>The selected track is displayed here.</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="fileBrowseButton">
        <property name="toolTip">
         <string>Browse File for Song</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../../resources.qrc">
          <normaloff>:/images/dialogs/browse.png</normaloff>:/images/dialogs/browse.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>32</width>
          <height>32</height>
         </size>
        </property>
        <property name="flat">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="playbackButton">
        <property name="toolTip">
         <string>Play</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../../resources.qrc">
          <normaloff>:/images/player/play.png</normaloff>:/images/player/play.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
        <property name="shortcut">
         <string>Ctrl+S</string>
        </property>
        <property name="flat">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="initializeButton">
        <property name="toolTip">
         <string>Initialize using the media meta data</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../../resources.qrc">
          <normaloff>:/images/dialogs/Initialize.png</normaloff>:/images/dialogs/Initialize.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
        <property name="flat">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="4" column="0" colspan="2">
    <widget class="QGroupBox" name="infoGroup">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="font">
      <font>
       <bold>true</bold>
      </font>
     </property>
     <property name="title">
      <string>Information</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="3" column="0">
       <widget class="QLabel" name="releasedYearLabel">
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Released Year</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
        </property>
        <property name="buddy">
         <cstring>releaseYearEdit</cstring>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="genreLabel">
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Genre</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
        </property>
        <property name="buddy">
         <cstring>genreEdit</cstring>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QSpinBox" name="releaseYearEdit">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimum">
         <number>1900</number>
        </property>
        <property name="maximum">
         <number>2100</number>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QComboBox" name="genreEdit">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="nameEdit">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="readOnly">
         <bool>false</bool>
        </property>
        <property name="clearButtonEnabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="artistLabel">
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Artist</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
        </property>
        <property name="buddy">
         <cstring>artistEdit</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="nameLabel">
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Name</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
        </property>
        <property name="buddy">
         <cstring>nameEdit</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="artistEdit">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="readOnly">
         <bool>false</bool>
        </property>
        <property name="clearButtonEnabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="6" column="0">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::StandardButton::Cancel|QDialogButtonBox::StandardButton::Ok</set>
     </property>
    </widget>
   </item>
   <item row="2" column="0" colspan="2">
    <widget class="ImageView" name="coverView" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="acceptDrops">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="3" column="0">
    <layout class="QHBoxLayout" name="removePhotoLayout">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <item>
      <spacer name="leftSpacer">
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="removePhotoButton">
       <property name="toolTip">
        <string>Remove Cover</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="../../resources.qrc">
         <normaloff>:/images/dialogs/delete.png</normaloff>:/images/dialogs/delete.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>32</width>
         <height>32</height>
        </size>
       </property>
       <property name="flat">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="rightSpacer">
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ImageView</class>
   <extends>QWidget</extends>
   <header>include/components/imageview.h</header>
   <container>1</container>
   <slots>
    <signal>viewActivated()</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>fileNameEdit</tabstop>
  <tabstop>fileBrowseButton</tabstop>
  <tabstop>nameEdit</tabstop>
  <tabstop>artistEdit</tabstop>
  <tabstop>genreEdit</tabstop>
  <tabstop>releaseYearEdit</tabstop>
 </tabstops>
 <resources>
  <include location="../../resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>fileBrowseButton</sender>
   <signal>clicked()</signal>
   <receiver>SongEdit</receiver>
   <slot>openSongFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>405</x>
     <y>73</y>
    </hint>
    <hint type="destinationlabel">
     <x>510</x>
     <y>10</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playbackButton</sender>
   <signal>clicked()</signal>
   <receiver>SongEdit</receiver>
   <slot>controlPlayback()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>455</x>
     <y>72</y>
    </hint>
    <hint type="destinationlabel">
     <x>523</x>
     <y>88</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>SongEdit</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>368</x>
     <y>755</y>
    </hint>
    <hint type="destinationlabel">
     <x>250</x>
     <y>648</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>SongEdit</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>444</x>
     <y>755</y>
    </hint>
    <hint type="destinationlabel">
     <x>487</x>
     <y>647</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>removePhotoButton</sender>
   <signal>clicked()</signal>
   <receiver>SongEdit</receiver>
   <slot>removeCover()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>255</x>
     <y>509</y>
    </hint>
    <hint type="destinationlabel">
     <x>326</x>
     <y>549</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>initializeButton</sender>
   <signal>clicked()</signal>
   <receiver>SongEdit</receiver>
   <slot>updateMetaData()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>482</x>
     <y>46</y>
    </hint>
    <hint type="destinationlabel">
     <x>523</x>
     <y>57</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>nameChanged(QString)</signal>
  <signal>artistChanged(QString)</signal>
  <signal>coverChanged(QPixmap)</signal>
  <signal>releasedYearChanged(int)</signal>
  <signal>genreChanged(Song::Genre)</signal>
  <slot>openSongFile()</slot>
  <slot>controlPlayback()</slot>
  <slot>removeCover()</slot>
  <slot>updateMetaData()</slot>
 </slots>
</ui>
